### PR TITLE
Create cli

### PR DESCRIPTION
**Pull request by magsteen**
Created CLI for fast testing with hyperparameter tweaking and model saving on the fly.

- Integrated with QTable
- NOT integrated with DQN yet

### CODE REVIEW

File: discretizer.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
 from vec import Vec2
 from math import ceil, pi
 import numpy as np
+from enum import Enum
 from snake_state import Direction
 
+class DISCRETIZER_TYPES(str, Enum):
```
Comment (chrisryda) [inline discretizer.py]: Enum names are singular in python, change to DISCRETIZER_TYPE

**Review by chrisryda — CHANGES_REQUESTED** [review]
Amazing work, looks really good to me. One small change needed, see my comment. This is not my area of expertise, so I don't feel comfortable approving for merge - Tommy or Norbert can do so:--)

**Review by chrisryda — APPROVED** [review]

File: main.py [hunk 2]
```diff
@@ -1,42 +1,138 @@
...
+    print(f"RENDER={render}")
+    print(f"SEED={seed}")
+    
+    agent_ctx = AgentContext(alpha, gamma, size)
+    environment_ctx = EnvironmentContext(size, render, seed)
+    
+    ctx.obj = CLIContext(agent_ctx, environment_ctx)
+
+@entry.command()
+@click.option("-d", "--discretizer", type=click.Choice(DISCRETIZER_TYPE), required=False, cls=OneOf, other="file")
```
Comment (tomrsae) [inline main.py:31]: needs to be an option

File: main.py [hunk 3]
```diff
@@ -1,42 +1,138 @@
...
+
+        if key.char == "s":
+            print("Saving current model state...")
+            file = agent.to_file()
+            print(f"Saved model state as \"{file}\".")
+            
+        elif key.char == "g":
+            print("Creating performance graph of current learning...")
+            # ...
+            print("Graph created and saved.")
```
Comment (tomrsae) [inline main.py]: print path to graph, like in file

File: utils/option_handlers.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            )
+                
+        return super(RequiredByWhenSetTo, self).handle_parse_result(ctx, opts, args)
+
+class OneOf(Option):
+    def __init__(self, *args, **kwargs):
+        self.other = kwargs.pop("other")
+        assert self.other, "'other' parameter required"
+        kwargs["help"] = (kwargs.get("help", "") +
+            f" NOTE: This option is mutually exclusive with the other option '{self.other}'"
```
Comment (tomrsae) [inline utils/option_handlers.py:33]: not accurate

**Review by tomrsae — CHANGES_REQUESTED** [review]

**Review by tomrsae — APPROVED** [review]
👍 😩

**Pull request merged**